### ISSUE TITLE
Saving best model and training statistics

### ISSUE THREAD

**Issue by oalee** [ilex-paraguariensis/yerbamate#1]
Saving best model, epochs, loss, best_loss, trained params, visualizations.

**Comment by SalamanderXing** [comment]
This feature should probably be probably implemented in the BaseModel (maybe called BaseMateModel)

**Comment by oalee** [comment]
Yes. We need to define a standard to return val_loss in all classes that extend BaseMateModel.
